circleguard/loadable.py

```diff
@@ -542,6 +544,24 @@ def __repr__(self):
     def __str__(self):
         return f"Replay by {self.username} on {self.map_id}"
 
+    def as_arrays(self):
+        """
+        Gets this replay's play data as arrays of absolute time, x and y, and pressed keys.
+
+        Returns
+        -------
+        ndarray[int], ndarray[[float]], ndarray[int]
+            The arrays of time, x and y, and pressed keys.
+        """
+        block = list(zip(*[(e.time_since_previous_action, e.x, e.y, e.keys_pressed) for e in self.replay_data]))
+
+        t = np.array(block[0], dtype=int)
+        t = t.cumsum()
+        xy = np.array([block[1], block[2]], dtype=float).T
+        k = np.array(block[3], dtype=int)
+
+        return t, xy, k
+
     def as_list_with_timestamps(self):
```

Comment:
Can this method be removed in favor of the new `t`, `xy`, `k`? Only `Investigator` uses it right now but I'm pretty sure it can be changed to use `ndarrays` instead of this method.